### DISCUSSION TITLE
How to controll LOAD switch by MQTT

### DISCUSSION THREAD

**Discussion by murfy76** [softwarecrash/EPEver2MQTT#11] [Q&A] [answered]
Hello,

I want to ask how to controll LOAD switch by MQTT command with JSON output. I'm trying to controll switch with "IoT MQTT Panel" android app. Thank you for help and also this great project!

Peter

**Comment by softwarecrash** [comment]
just change the value of the LOAD_STATE

**Comment by murfy76** [comment]
Sure, but command mosquitto_pub -t solar/epever/DATA -m {\"LOAD_STATE\":\"false\"} is not working. What's I'm missing? Thanks.

**Comment by softwarecrash** [comment, marked as answer]
so first it is a bool value not a text, the "" not needed
{"LOAD_STATE": true}
have you more than one device connected or set in the settings?

**Reply by murfy76** [reply]
You are right, it's working now:  

mosquitto_pub -t solar/epever/DATA -m {\"LOAD_STATE\":false}
mosquitto_pub -t solar/epever/DATA -m {\"LOAD_STATE\":true}

Thanks!

**Comment by TronTronix** [comment]
Hallo,
Irgendwie habe ich ein Verständnis Problem mit der Ansteuerung des LOAD Ausgangs.
Über das Web Interface kann ich Load State An und Ausschalten, soweit also OK.
Über MQTT also mein Pfad `EPEver/EP_1/LiveData/LOAD_STATE` 
Ändere ich in true
Das wird kurz angezeigt und wird von Programm auf false einfach zurückgesetzt.

Bei MQTT Json Style also so
`mosquitto_pub -t EPEver/DATA -m {"LOAD_STATE":true}`
wird es zwar angezeigt 
`{LOAD_STATE:true}`
Aber das Licht geht nicht an und bei der Aktualisierung wider false
Was mache ich falsch ??

**Reply by softwarecrash** [reply]
schau dir mal bitte das aktuelle json an, da ist es nummeriert, also EP_1
```
{
   "EP_1":{
      "LiveData":{
         "LOAD_STATE":false
      },
 },
}
```

**Reply by TronTronix** [reply]
Ja genau das mache ich mit (MQTT Explorer) das ist die Stelle wo ich es ändere.
In diesen fall auf den Konsole des Mosquitto, mein Befehl ist dafür
 `mosquitto_pub -t EPEver/EP_1/LiveData/LOAD_STATE -m true`

auf (MQTT Explorer) kann ich es sehen es wird kurz angezeigt aber es scheint es wird nicht übermittelt zum ESP und er setzt ein false bei der nächsten Aktualisierung rein.

**Reply by softwarecrash** [reply]
nicht durcheinander bringen...
wenn du mqtt json style eingestellt  hast dann:
`mosquitto_pub -t EPEver/EP_1/DATA {"EP_1":{"LiveData":{"LOAD_STATE":true}}}`
wenn du nur mqtt hast dann:
`mosquitto_pub -t EPEver/EP_1/DeviceControl/LOAD_STATE -m true`

EDIT:
im json style ist tatsächlich aktuell ein bug drin

EDIT2:
ist mit V2.0.2 Behoben.
geht jetzt mit den obigen beispielen, füge es eben noch ins wiki ein

**Reply by TronTronix** [reply]
Ja das war es auf den Pfad wäre ich nie in Leben gekommen außer ich hätte den Quellcode gelesen in der Datei main.cpp. Danke

**Reply by softwarecrash** [reply]
ich vergesse immer das es bei anderen nicht angezeigt wird.
IOB zb legt einen DP an wenn ein pfad ein subscribe bekommt

**Reply by TronTronix** [reply]
So habe die neue Firmware drauf, ist aber nicht so wie du schreibst bei **mqtt json style** sondern so.
`mosquitto_pub -t EPEver/DATA {"EP_1":{"LiveData":{"LOAD_STATE":true}}}`
Leider ist das genau womit ich arbeite, und warum müßen wir die gleiche variabel benutzen um Daten zu Senden und zu empfangen.
Das gibt wiederum bei der Datenabfrage aus den EPEver.... das mitunter die Daten nicht verfügbar sind durch den LOAD_STATE Befehl.

**Reply by softwarecrash** [reply]
> So habe die neue Firmware drauf, ist aber nicht so wie du schreibst bei **mqtt json style** sondern so.
> `mosquitto_pub -t EPEver/DATA {"EP_1":{"LiveData":{"LOAD_STATE":true}}}`
> Leider ist das genau womit ich arbeite, und warum müßen wir die gleiche variabel benutzen um Daten zu Senden und zu empfangen.
> Das gibt wiederum bei der Datenabfrage aus den EPEver.... das mitunter die Daten nicht verfügbar sind durch den LOAD_STATE Befehl.
> 
> 

Das ist ein übrigbleibsel aus früheren Zeiten, ich benutzte kein json, hatte das nur Mal auf Nachfrage eingebaut.
Ich bin da für lösungsvorschläge offen.
Man könnte auch den Pfad wie bei normalen mqtt beibehalten und ein Control nutzen.

**Reply by TronTronix** [reply]
Ich wollte nur die Problematik von  (_Json String over MQTT_) offenlegen, Lösungsvorschläge kann ich jetzt noch nicht viel zu sagen bin erstmals auf IP/livejson umgestiegen da gibt es solche Störungen nicht und ich greife bei der Abfrage direkt ohne dritte auf den ESP zu, was aber bei Statistiken eine Lachnummer wird.

**Reply by SpeedyKR** [reply]
Hallo TronTronix,
ich habe auch die Probleme mit dem Load_Controll gehabt, da ich ihn mit Node-red schalten wollte.
Habe die gleiche Erfahrung gemacht, das ein schicken von   {LOAD_STATE:true} auf den MQTT Topic
EPEver/DATA , das der Load Pin nur kurz aus geschaltet wurde aber keine Funktion am Eperver, nur 
auf meinem dashboard wurde kurz umgeschaltet und nach kurzer zeit wieder zurück geschaltet.
ich habe dann mir den  MQTT-Explorer installiert und die Komplette MQTT Strucktur anzeigen lassen.
Da sieht man , das  Unter Puplish   Topic   EPEver/DATA   und als Payload  {LOAD_STATE:true} 
 geschaltet wird und mit {LOAD_STATE:false} wieder aus geschaltet wird.
Aber vorsetzung ist das im Epever Regler, der Timer ausgeschaltet ist .
Ich habe dann bei Node-red das ganze über einen Taster und MQTT Ausgang  geschaltet.
Der MQTT-Explorer ist eine grosse hilfe beim suchen von MQTT Struckturen .....
Hoffe den einen oder anderen einen Hinweis gegeben zuhaben..

**Reply by SpeedyKR** [reply]
Hallo zusammen...
als kleine Anmerkung , 
LOAD_STATE wird jetzt , seit version 2.03 mit neuer MQTT Strucktur .
Bei meinen Einstellungen : 
Device Name  : EPEver2MQTT
MQTT Topic    :  EPEver
MQTT Data Trigger Path  :  EPEver/DATA
folgender massen geschaltet : 
Über MQTT.fx 
Publish :  EPEver/DATA
Payload      : {"EP_1":{"LiveData":{"LOAD_STATE":true}}}  // Eingeschaltet
Payload      : {"EP_1":{"LiveData":{"LOAD_STATE":false}}}  // Ausgeschaltet

Das ganze im Node_red mit einem Button, über Dashboard .

Payload 
{"EP_1":{"LiveData":{"LOAD_STATE":true}}}  // Eingeschaltet
{"EP_1":{"LiveData":{"LOAD_STATE":false}}}  // Ausgeschaltet

falls der eine oder andere Probleme mit dem Schalten des LOAD_STATE hat .
gruß Speedy
